File: iru/week1/easy.js
function solution(nums) {
  let pickLimit = Math.floor(nums.length / 2);
  let empty = [];
  let result = 0;

  for (let i = 0; i < nums.length; i++) {
    let currentType = nums[i];

    if (empty[currentType] === undefined) {
      if (result < pickLimit) {
        result += 1;
      }
      empty[currentType] = true;
    }
  }

  return result;
}

File: jimin/week1/easy.py
def solution(nums):
    # 고를 수 있는 마리 수
    pick = len(nums) // 2

    # 서로 다른 종류 수
    kinds = len(set(nums))

    # 종류를 최대한 많이 고르려면
    # (서로 다른 종류 수)와 (고를 수 있는 마리 수) 중 작은 값이 정답
    return min(kinds, pick)
